src/simsopt/geo/curveobjectives.py

```diff
@@ -242,3 +249,102 @@ def dJ(self):
+        arclength. The key question is now the choice of :math:`l`. Enforcing zero
+        variance of :math:`l` value essentially corresponds to :math:`l-1`
+        constraints. If we pick :math:`l` too small, the objective is still
+        ill-posed, if we pick the value to high, then the constraint is too strong
+        and limits what sort of curves we can represent.
+
+        The argument blow provides some intuition for the choice of :math:`l`. A
+        curve in 3d space is defined uniquely by an initial point, an initial
+        direction, and the arclength, curvature, and torsion along the curve. For a
+        :mod:`simsopt.geo.curvexyzfourier.CurveXYZFourier`, the intuition is now as
+        follows: assuming that the curve has order :math:`p`, that means we have
+        :math:`3*(2p+1)` degrees of freedom in total. Assuming that three each are
+        required for both the initial position and direction, :math:`6p-3` are left
+        over for curvature, torsion, and arclength. We want to fix the arclength,
+        so we can afford :math:`2p-1` constraints, which corresponds to
+        :math:`l=2p`.
+        """
+        super().__init__(depends_on=[curve])
+        self.curve = curve
+        nquadpoints = len(curve.quadpoints)
+        if nintervals is None:
+            from simsopt.geo.curvexyzfourier import CurveXYZFourier, JaxCurveXYZFourier
+            if isinstance(curve, CurveXYZFourier) or isinstance(curve, JaxCurveXYZFourier):
+                nintervals = 2*curve.order
+            else:
+                raise RuntimeError("Please provide a value for `nintervals`. We only have a default for `CurveXYZFourier` and `JaxCurveXYZFourier`.")
+        self.nintervals = nintervals
+        indices = np.floor(np.linspace(0, nquadpoints, nintervals+1, endpoint=True)).astype(int)
+        mat = np.zeros((nintervals, nquadpoints))
+        for i in range(nintervals):
+            mat[i, indices[i]:indices[i+1]] = 1/(indices[i+1]-indices[i])
+        self.mat = mat
+        self.thisgrad = jit(lambda l: grad(lambda x: curve_arclengthvariation_pure(x, mat))(l))
+
+    def J(self):
+        return float(curve_arclengthvariation_pure(self.curve.incremental_arclength(), self.mat))
+
+    @derivative_dec
+    def dJ(self):
+        """
+        This returns the derivative of the quantity with respect to the curve dofs.
+        """
+        return self.curve.dincremental_arclength_by_dcoeff_vjp(
+            self.thisgrad(self.curve.incremental_arclength()))
+
+    return_fn_map = {'J': J, 'dJ': dJ}
+
+
+@jit
+def curve_msc_pure(kappa, gammadash):
+    """
+    This function is used in a Python+Jax implementation of the curve arclength variation.
+    """
+    arc_length = jnp.linalg.norm(gammadash, axis=1)
+    return jnp.mean(kappa**2 * arc_length)/jnp.mean(arc_length)
+
+
+class MeanSquaredCurvature(Optimizable):
+
+    def __init__(self, curve):
+        r"""
+        Compute the mean of the squared curvature of a curve.
```

Comment:
Can we include an explicit formula for `J` here in the docstring, e.g. `(1/L) \int d\ell \kappa^2` where `L` is the curve length and `d\ell` is the incremental arclenth?